ts/Extensions/Boost/BoostSeries.ts

```diff
@@ -849,6 +859,111 @@ function getPoint(
     return point;
 }
 
+/**
+ * @private
+ */
+function scatterProcessData(
+    this: Series,
+    force?: boolean
+): (boolean|undefined) {
+    const series = this,
+        options = series.options,
+        xAxis = series.xAxis,
+        yAxis = series.yAxis;
+    (window as any).TheSeries = series;
+
+    // Process only on changes
+    if (
+        !series.isDirty &&
+        !xAxis.isDirty &&
+        !yAxis.isDirty &&
+        !force
+    ) {
+        return false;
+    }
+
+    // Required to get tick-based zoom ranges
+    series.yAxis.setTickInterval();
+
+    const boostThreshold = options.boostThreshold || 0,
+        cropThreshold = options.cropThreshold,
+        data = options.data || series.data,
+        xData = series.xData as Array<number>,
+        xExtremes = xAxis.getExtremes(),
+        xMax = xExtremes.max,
+        xMin = xExtremes.min,
+        yData = series.yData as Array<number>,
+        yExtremes = yAxis.getExtremes(),
+        yMax = yExtremes.max,
+        yMin = yExtremes.min;
+
+    // Skip processing in non-boost zoom
+    if (
+        !series.boosted &&
+        xAxis.old &&
+        yAxis.old &&
+        xMin >= (xAxis.old.min || 0) &&
```

Comment:
This looks suspicious: `(xAxis.old.min || 0)`. What if we're dealing with subzero values? Since the max falls back to `Number.MAX_VALUE`, I suspect the min should fall back to `-Number.MAX_VALUE`.

Also, a simple `||` operator may not be safe, because if `min` or `max` are the number 0, it will count as falsy. I think what we want to use is the `??` operator.

Reply:
Note the difference between `Number.MIN_VALUE` and `-Number.MAX_VALUE`. `Number.MIN_VALUE` is the least possible number _above zero_.

Reply:
Thank you. It's always a trap for me.